clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -11285,6 +11285,21 @@ Value *CodeGenFunction::EmitAArch64SMEBuiltinExpr(unsigned BuiltinID,
   if (Builtin->LLVMIntrinsic == 0)
     return nullptr;
 
+  if (BuiltinID == SME::BI__builtin_sme___arm_in_streaming_mode) {
+    // If we already know the streaming mode, don't bother with the intrinsic
+    // and emit a constant instead
```

Comment:
(optional suggestion): Rather than doing this while emitting builtins you could always emit the intrinsic, then handle the `Intrinsic::aarch64_sme_in_streaming_mode` in `llvm/lib/Analysis/ConstantFolding.cpp` (other target specific intrinsics are already handled there).